pkg/isoeditor/nmstate_handler.go

```diff
@@ -0,0 +1,114 @@
+package isoeditor
+
+import (
+	"bytes"
+	"fmt"
+	"os"
+	"os/exec"
+	"path/filepath"
+	"regexp"
+	"strings"
+
+	"github.com/pkg/errors"
+	log "github.com/sirupsen/logrus"
+)
+
+//go:generate mockgen -package=isoeditor -destination=mock_nmstate_handler.go . NmstateHandler
+type NmstateHandler interface {
+	CreateNmstateRamDisk(nmstatectlPath, ramDiskPath string) error
+	ExtractNmstatectl(rootfsPath string) (string, error)
+}
+
+type nmstateHandler struct {
+	workDir, nmstateDir string
+}
+
+func NewNmstateHandler(workDir string) NmstateHandler {
+	return &nmstateHandler{
+		workDir: workDir,
+	}
+}
+
+func (n *nmstateHandler) CreateNmstateRamDisk(nmstatectlPath, ramDiskPath string) error {
+	// Check if nmstatectl binary file exists
+	if _, err := os.Stat(nmstatectlPath); os.IsNotExist(err) {
+		return err
+	}
+
+	// Read binary
+	nmstateBinContent, err := os.ReadFile(nmstatectlPath)
+	if err != nil {
+		return err
+	}
+
+	// Create a compressed RAM disk image with the nmstatectl binary
+	compressedCpio, err := generateCompressedCPIO(nmstateBinContent, NmstatectlPathInRamdisk, 0o100_755)
+	if err != nil {
+		return err
+	}
+
+	// Write RAM disk file
+	err = os.WriteFile(ramDiskPath, compressedCpio, 0755) //nolint:gosec
+	if err != nil {
+		return err
+	}
+
+	// Remove temp dir
+	if err := os.RemoveAll(n.nmstateDir); err != nil {
+		return err
+	}
+
+	return nil
+}
+
+func (n *nmstateHandler) ExtractNmstatectl(rootfsPath string) (string, error) {
+	var err error
+	n.nmstateDir, err = os.MkdirTemp(n.workDir, "nmstate")
```

Comment:
This doesn't feel like it should be state.

This makes the object not thread safe and also looks like this will leak this directory on error in this function or `CreateNmstateRamDisk`